pandas/core/arrays/categorical.py

```diff
@@ -532,7 +533,12 @@ def astype(self, dtype: AstypeArg, copy: bool = True) -> ArrayLike:
         else:
             # GH8628 (PERF): astype category codes instead of astyping array
             try:
-                new_cats = np.asarray(self.categories)
+                if is_datetime64_dtype(self.categories):
+                    values = ensure_wrapped_if_datetimelike(np.asarray(self.categories))
```

Comment:
`self.categories._values` should be what you want here.  thats probably also an improvement if `is_datetime64tz_dtype(self.categories.dtype)`

Reply:
tz aware datetimes are alreday handled correctly

Reply:
i was thinking it might be a perf improvement for dt64tz, but either way its fine to consider that out of scope for this PR